src/Nethermind/Nethermind.Evm/Instructions/EvmInstructionsUtils.cs

```diff
@@ -0,0 +1,229 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Int256;
+
+namespace Nethermind.Evm;
+
+
+
+
+public static class EvmInstructionsUtils
```

Comment:
Maybe better naming? Open to other suggestions.

```suggestion
public static class EvmCalculations
```